Create ComboBox from shadcn example

dewert99 opened the pull request:
Another attempt at #934 since we're not trying to replicate MUI.

TODO:
Handle uses of ComboBox outside `platform-bible-react`

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/paranext/paranext-core/970)
<!-- Reviewable:end -->

jolierabideau commented on lib/platform-bible-react/src/components/combo-box.component.tsx:
```diff
@@ -55,67 +23,86 @@ export type ComboBoxProps<T> = {
+    <Popover open={open} onOpenChange={setOpen}>
+      <PopoverTrigger asChild>
+        <Button
+          variant="outline"
```
This should be an optional prop, `buttonVariant` that defaults to "outline"

jolierabideau commented on lib/platform-bible-react/src/components/combo-box.component.tsx:
```diff
@@ -55,67 +23,86 @@ export type ComboBoxProps<T> = {
+      <PopoverContent className="pr-w-[200px] pr-p-0">
+        <Command>
+          <CommandInput placeholder={testPlaceholder} className="pr-text-inherit" />
+          <CommandEmpty>No framework found.</CommandEmpty>
```
This should be an optional prop, `commandEmptyMessage` or something

jolierabideau commented on lib/platform-bible-react/src/components/combo-box.component.tsx:
```diff
@@ -55,67 +23,86 @@ export type ComboBoxProps<T> = {
+                <Check
+                  className={cn(
+                    'pr-mr-2 pr-h-4 pr-w-4',
+                    !value || getOptionLabel(value) === getOptionLabel(option)
+                      ? 'pr-opacity-100'
+                      : 'pr-opacity-0',
```
I am still learning about the `cn` function, and I think this solution works, but it is cleaner to use an object to define the conditional. `{ 'pr-opacity-100': !value || getOptionLabel(value) === getOptionLabel(option) }` An example is in the `TextField` component

jolierabideau commented on lib/platform-bible-react/src/components/combo-box.component.tsx:
```diff
@@ -55,67 +23,86 @@ export type ComboBoxProps<T> = {
+}
+
 /**
  * Dropdown selector displaying various options from which to choose
```
Just a small thing but we can make this description the one from the Shadcn page "Autocomplete input and command palette with a list of suggestions."

jolierabideau commented on lib/platform-bible-react/src/components/ref-selector.component.tsx:
```diff
@@ -14,6 +14,7 @@ import '@/components/ref-selector.component.css';
 import ComboBox, { ComboBoxLabelOption } from '@/components/combo-box.component';
 import Button from '@/components/button.component';
 import TextField from '@/components/text-field.component';
+import { Label } from './shadcn-ui/label';
```
aliased path here

jolierabideau commented on lib/platform-bible-react/src/components/ref-selector.component.tsx:
```diff
@@ -50,11 +51,11 @@ function RefSelector({ scrRef, handleSubmit, id }: ScrRefSelectorProps) {
 
-  const onSelectBook = (_event: SyntheticEvent<Element, Event>, value: unknown) => {
+  const onSelectBook = (value: BookNameOption) => {
     // Asserting because value is type unknown, value is type unknown because combobox props aren't precise enough yet
     // Issue https://github.com/paranext/paranext-core/issues/560
     // eslint-disable-next-line no-type-assertion/no-type-assertion
```
Can remove these

jolierabideau commented on lib/platform-bible-react/src/components/combo-box.component.tsx:
```diff
@@ -1,50 +1,18 @@
-export type ComboBoxChangeDetails<T> = AutocompleteChangeDetails<T>;
-export type ComboBoxChangeReason = AutocompleteChangeReason;
 
 export type ComboBoxProps<T> = {
```
I don't think we need the generic <T> here anymore

jolierabideau commented on lib/platform-bible-react/src/components/combo-box.component.tsx:
Are you still working on the styling? It seems weird that the fonts aren't going to the book list. If you run into problems with fixing this let me know

From preview page:
![Screenshot 2024-06-28 at 9 13 53 AM](https://github.com/paranext/paranext-core/assets/135999578/7cd4a95e-a8df-4960-bf11-1c06fb0e363f)

From Shadcn website:
![Screenshot 2024-06-28 at 9 13 41 AM](https://github.com/paranext/paranext-core/assets/135999578/fb47ec92-fa35-45a3-8cea-cf9e78a7ed65)

Sebastian-ubs replied:
Hi @dewert99, what did you have to do to make the styles / font apply also elsewhere?
I remember I had to manually add the font somehwere in the preview app, because the preflight styles made it no longer apply to the whole page.
see https://github.com/paranext/paranext-core/pull/896/files#diff-6caee5bf8f5130b276e0d6adbc0484f0cfeb541d535975ae6fa5a2e402c1e9c6R13

jolierabideau reviewed (changes requested):
Thanks for your work on this David! It's looking good so far

dewert99 commented:
*[`lib/platform-bible-react/src/components/combo-box.component.tsx` line 0 at r1](https://reviewable.io/reviews/paranext/paranext-core/970#gh-1658743316:-O12ay9Q4wmRxjqYQToM:b-nl23s4) ([raw file](https://github.com/paranext/paranext-core/blob/08d0b5c2f8802e811b5474f071c0d0603fd5b25d/lib/platform-bible-react/src/components/combo-box.component.tsx#L0)):*
<details><summary><i>Previously, Sebastian-ubs wrote…</i></summary><blockquote>

Hi @dewert99, what did you have to do to make the styles / font apply also elsewhere?
I remember I had to manually add the font somehwere in the preview app, because the preflight styles made it no longer apply to the whole page.
Added to the app https://github.com/paranext/paranext-core/pull/896/files#diff-6caee5bf8f5130b276e0d6adbc0484f0cfeb541d535975ae6fa5a2e402c1e9c6R13
and was also needed to be addded to the dropdown menu in addition
https://github.com/paranext/paranext-core/pull/896/files#diff-593ade974b2291d4db98c6a56d290de4dd0ad56861711f880c2d78b641d8ccebR65
</blockquote></details>

I don't think I changed the files `shad-cn` added at all. I kept the styles in `ComboBox` pretty much the same as in the `shad-cn` example, except that I added `pr-text-inherit` to the `CommandInput` so the text the user types would be visible.


<!-- Sent from Reviewable.io -->

Sebastian-ubs reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: 0 of 14 files reviewed, 12 unresolved discussions (waiting on @jolierabideau)

___
*[`lib/platform-bible-react/src/components/combo-box.component.tsx` line 0 at r1](https://reviewable.io/reviews/paranext/paranext-core/970#gh-1658743316:-O1M4DfB6OV9dd3k5Upg:bbpe00m) ([raw file](https://github.com/paranext/paranext-core/blob/08d0b5c2f8802e811b5474f071c0d0603fd5b25d/lib/platform-bible-react/src/components/combo-box.component.tsx#L0)):*
<details><summary><i>Previously, dewert99 (David Ewert) wrote…</i></summary><blockquote>

I don't think I changed the files `shad-cn` added at all. I kept the styles in `ComboBox` pretty much the same as in the `shad-cn` example, except that I added `pr-text-inherit` to the `CommandInput` so the text the user types would be visible.
</blockquote></details>

Interesting. Thank you. When I am trying to just use the dropdown as is from shadcn, in the preview app I am getting unstyled resuts. See
![grafik.png](https://files.reviewableusercontent.io/images/reviewable/141921979/Ysxo3OysTDY3uzxjoIab5jWc/grafik.png)


<!-- Sent from Reviewable.io -->

jolierabideau reviewed (commented):
Thanks David! You might want to pull the recent core because I noticed some differences. If you wouldn't mind using your combobox in core and in the preview page, and attach screenshots to this PR description that would be great! You can add your combobox to the preview page in `basics.component.tsx` in its own tab so that people can see/play with it when you merge. 

> Reviewed 6 of 14 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970#-:-O1YABur2dxit9VjbOd9:b8ejkrv)* status: all files reviewed, 12 unresolved discussions (waiting on @dewert99)

___
*[`lib/platform-bible-react/src/components/combo-box.component.tsx` line 7 at r1](https://reviewable.io/reviews/paranext/paranext-core/970#gh-1658706163:-O1Y7A3R-O9Uq8e8nlhH:bsm9svy) ([raw file](https://github.com/paranext/paranext-core/blob/08d0b5c2f8802e811b5474f071c0d0603fd5b25d/lib/platform-bible-react/src/components/combo-box.component.tsx#L7)):*
<details><summary><i>Previously, dewert99 (David Ewert) wrote…</i></summary><blockquote>

Done. Would it be worth having a lint to help remember this?
</blockquote></details>

I think that is a good idea, I will mention it and see if the team agrees
___
*[`lib/platform-bible-react/src/components/combo-box.component.tsx` line 41 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YBBfXEl2iwxog_t5r:-O1YBBfXEl2iwxog_t5s:b-qslhk0) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/combo-box.component.tsx#L41)):*
> ```XML
>   commandEmptyMessage?: string;
>   /** Variant of button */
>   buttonVariant?: 'outline' | 'link' | 'default' | 'secondary' | 'destructive' | 'ghost';
> ```

Just curious- is there a way to get these from the `buttonVariants` object in the button shadcn component? So that if we add more later we don't need to edit multiple places
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRUNbCQYm8hagrKUa:-O1YRUNc1PLyxIw6X6j2:b4mw41m) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L1)):*
> ```XML
> import * as React from 'react';
> ```

Have you tried to build `platform-bible-react`? I know this line causes problems, you may need to change it to `import React from 'react'`
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 26 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YS1w23FjRa394sW25:-O1YS1w23FjRa394sW26:b8mu1py) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L26)):*
> ```XML
> interface CommandDialogProps extends DialogProps {}
> 
> const CommandDialog = ({ children, ...props }: CommandDialogProps) => {
> ```

I am seeing a lint error on this function,
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 42 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSAkIDQPlVAihz1So:-O1YSAkIDQPlVAihz1Sp:b-rz2n85) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L42)):*
> ```XML
>   React.ComponentPropsWithoutRef<typeof CommandPrimitive.Input>
> >(({ className, ...props }, ref) => (
>   <div className="pr-flex pr-items-center pr-border-b pr-px-3" cmdk-input-wrapper="">
> ```

Getting a lint error on `cmdx-input-wrapper`
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 123 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSFq8BZv7MQAbDSRu:-O1YSFq8BZv7MQAbDSRv:b1cq98u) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L123)):*
> ```XML
> CommandItem.displayName = CommandPrimitive.Item.displayName;
> 
> const CommandShortcut = ({ className, ...props }: React.HTMLAttributes<HTMLSpanElement>) => {
> ```

Getting a lint error on this function
___
*[`lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRfCADrRbqSd3ytVz:-O1YRfCB1j20YE7f4to5:bbxgkzp) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx#L1)):*
> ```XML
> import * as React from 'react';
> ```

Same comment as above
___
*[`lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx` line 54 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSK350-HrSlUujyyO:-O1YSK350-HrSlUujyyP:b1y2m21) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx#L54)):*
> ```XML
> DialogContent.displayName = DialogPrimitive.Content.displayName;
> 
> const DialogHeader = ({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) => (
> ```

Lint error on this function
___
*[`lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx` line 62 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSLU5CU4quqvPWqgO:-O1YSLU5CU4quqvPWqgP:b1y2m21) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx#L62)):*
> ```XML
> DialogHeader.displayName = 'DialogHeader';
> 
> const DialogFooter = ({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) => (
> ```

Lint error on this function
___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRl3eBT7RWqjuLkom:-O1YRl3eBT7RWqjuLkon:bbxgkzp) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L1)):*
> ```XML
> import * as React from 'react';
> ```

Same comment as above


<!-- Sent from Reviewable.io -->

Sebastian-ubs reviewed (commented):
please be aware with a little code structure change, moving stuff from `basic.component.tsx` into separate components:
https://github.com/paranext/paranext-core/pull/997

> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970#-:-O1bHkl0C4UrxlVMMKX9:bgprl31)* status: all files reviewed, 12 unresolved discussions (waiting on @dewert99 and @jolierabideau)

___
*[`lib/platform-bible-react/src/components/combo-box.component.tsx` line 0 at r1](https://reviewable.io/reviews/paranext/paranext-core/970#gh-1658743316:-O1MBiQP6fQyd_4X-yZj:bbv1npd) ([raw file](https://github.com/paranext/paranext-core/blob/08d0b5c2f8802e811b5474f071c0d0603fd5b25d/lib/platform-bible-react/src/components/combo-box.component.tsx#L0)):*
<details><summary><i>Previously, Sebastian-ubs wrote…</i></summary><blockquote>

Interesting. Thank you. When I am trying to just use the dropdown as is from shadcn, in the preview app I am getting unstyled resuts. See
![grafik.png](https://files.reviewableusercontent.io/images/reviewable/141921979/Ysxo3OysTDY3uzxjoIab5jWc/grafik.png)
</blockquote></details>

Found that the issue is that the dropdown content is added to the dom at a high level as sibling to the app "root" and so does not get all styling done by the app - theming however works as this goes up to the html root.
Problem was that therefore .pr-twp is not added to the dropdown content, which I will now fix in 


<!-- Sent from Reviewable.io -->

dewert99 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: 5 of 14 files reviewed, 12 unresolved discussions (waiting on @jolierabideau)

___
*[`lib/platform-bible-react/src/components/combo-box.component.tsx` line 41 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YBBfXEl2iwxog_t5r:-O1hMCZz2wN-OGDK2JCP:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/combo-box.component.tsx#L41)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Just curious- is there a way to get these from the `buttonVariants` object in the button shadcn component? So that if we add more later we don't need to edit multiple places
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRUNbCQYm8hagrKUa:-O1gvOBZ-PCSLj1i-LCl:bjo4s5m) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L1)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Have you tried to build `platform-bible-react`? I know this line causes problems, you may need to change it to `import React from 'react'`
</blockquote></details>

I did and I saw the lint errors, as well as `src/components/shadcn-ui/tabs-vertical.tsx (1:0) Error when using sourcemap for reporting an error: Can't resolve original location of error.` but I didn't see this error. I could still make the changes if you would like.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 26 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YS1w23FjRa394sW25:-O1guL8v8A0X0E_skZem:b-w1oph0) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L26)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

I am seeing a lint error on this function,
</blockquote></details>

Should I fix these lint errors, or should I disable them for files in `shadcn-ui` since they are auto generated?


<!-- Sent from Reviewable.io -->

jolierabideau reviewed (commented):
> Reviewed 8 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: 13 of 14 files reviewed, 11 unresolved discussions (waiting on @dewert99)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRUNbCQYm8hagrKUa:-O1rHR6X297xxYtkgDVb:b-9ikdhf) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L1)):*
<details><summary><i>Previously, dewert99 (David Ewert) wrote…</i></summary><blockquote>

I did and I saw the lint errors, as well as `src/components/shadcn-ui/tabs-vertical.tsx (1:0) Error when using sourcemap for reporting an error: Can't resolve original location of error.` but I didn't see this error. I could still make the changes if you would like.
</blockquote></details>

I see the tabs-vertical error, you can ignore that we'll look into it. Interesting that you didn't see this, you could go ahead and fix it
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 26 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YS1w23FjRa394sW25:-O1rHdvjCZO1fRpkQ-H1:bj7a3ut) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L26)):*
<details><summary><i>Previously, dewert99 (David Ewert) wrote…</i></summary><blockquote>

Should I fix these lint errors, or should I disable them for files in `shadcn-ui` since they are auto generated?
</blockquote></details>

We decided we would fix the lint errors in these files because we should only need to do it once when we add the component. A lot of them are auto-fixable. 


<!-- Sent from Reviewable.io -->

dewert99 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: 2 of 15 files reviewed, 11 unresolved discussions (waiting on @jolierabideau)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRUNbCQYm8hagrKUa:-O1sYgBU3mCX4mnOsLMo:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L1)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

I see the tabs-vertical error, you can ignore that we'll look into it. Interesting that you didn't see this, you could go ahead and fix it
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 26 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YS1w23FjRa394sW25:-O1sYibn2Yc7YpkDbwTI:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L26)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

We decided we would fix the lint errors in these files because we should only need to do it once when we add the component. A lot of them are auto-fixable.
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 42 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSAkIDQPlVAihz1So:-O1sYkIh6JBT8bmVuaYe:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L42)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Getting a lint error on `cmdx-input-wrapper`
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/command.tsx` line 123 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSFq8BZv7MQAbDSRu:-O1sYlbC8qP9se72-KIo:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/command.tsx#L123)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Getting a lint error on this function
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRfCADrRbqSd3ytVz:-O1sYnEA4Q9lbAjHCqaO:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx#L1)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Same comment as above
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx` line 54 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSK350-HrSlUujyyO:-O1sYoJQBPFRvd1SQlPY:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx#L54)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Lint error on this function
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx` line 62 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YSLU5CU4quqvPWqgO:-O1sYpN56NNOVGlvR3H9:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/dialog.tsx#L62)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Lint error on this function
</blockquote></details>

Done.
___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 1 at r2](https://reviewable.io/reviews/paranext/paranext-core/970#-O1YRl3eBT7RWqjuLkom:-O1sZ5TmCvTKyliTjeXa:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/99b5bdf6fe01fc690c9d8c7feb1028ee498ef8f5/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L1)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Same comment as above
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

jolierabideau reviewed (commented):
It's looking awesome David, thanks for your work on this! I think there are some more changes on core that you need to pull. Once you pull those, you can add the combobox to the preview page basic components tab!

> Reviewed 1 of 9 files at r3, 12 of 12 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970#-:-O1vinDtAniQnQAEwyWL:bgqngja)* status: all files reviewed, 1 unresolved discussion



<!-- Sent from Reviewable.io -->

jolierabideau reviewed (commented):
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: all files reviewed, 3 unresolved discussions (waiting on @dewert99)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O1wfHPhCm1TaLIHsX75:bjkhs5d) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
> ```XML
>       sideOffset={sideOffset}
>       className={cn(
>         'pr-z-50 pr-w-72 pr-rounded-md pr-border pr-bg-popover pr-p-4 pr-text-popover-foreground pr-shadow-md pr-outline-none data-[state=open]:pr-animate-in data-[state=closed]:pr-animate-out data-[state=closed]:pr-fade-out-0 data-[state=open]:pr-fade-in-0 data-[state=closed]:pr-zoom-out-95 data-[state=open]:pr-zoom-in-95 data-[side=bottom]:pr-slide-in-from-top-2 data-[side=left]:pr-slide-in-from-right-2 data-[side=right]:pr-slide-in-from-left-2 data-[side=top]:pr-slide-in-from-bottom-2',
> ```

I'm seeing that styles aren't being applied to the popover, can you add `pr-twp` to the front of this `className` string and see if that fixes the styling?
___
*[`lib/platform-bible-react/src/preview/components/basics/combo-box.examples.component.tsx` line 7 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfdJm07p2PHvD5X5d:-O1wfdJm07p2PHvD5X5e:b-92m0in) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/preview/components/basics/combo-box.examples.component.tsx#L7)):*
> ```XML
>   const [comboBoxValue, setComboBox] = useState<string | undefined>(undefined);
>   return (
>     <ComboBox
> ```

Can you see if you can adjust the height of the input field inside of the popover? Seems taller than the other inputs in the preview page


<!-- Sent from Reviewable.io -->

dewert99 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: 12 of 17 files reviewed, 3 unresolved discussions (waiting on @jolierabideau)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O1xJF27Dvbu5jRdcdDe:b3xusuu) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

I'm seeing that styles aren't being applied to the popover, can you add `pr-twp` to the front of this `className` string and see if that fixes the styling?
</blockquote></details>

I tried this and it seemed to fix the height of the text field, I'm not sure what other styling you were hoping to fix.
___
*[`lib/platform-bible-react/src/preview/components/basics/combo-box.examples.component.tsx` line 7 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfdJm07p2PHvD5X5d:-O1xJRG94KeBjoh8_leZ:b-yes80e) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/preview/components/basics/combo-box.examples.component.tsx#L7)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Can you see if you can adjust the height of the input field inside of the popover? Seems taller than the other inputs in the preview page
</blockquote></details>

This was fixed by using `pr-twp` in `popover.tsx`


<!-- Sent from Reviewable.io -->

jolierabideau reviewed (commented):
> Reviewed 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: all files reviewed, 2 unresolved discussions (waiting on @dewert99)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O2-e-yo3FzT9N7QIqKv:b-891c6p) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
<details><summary><i>Previously, dewert99 (David Ewert) wrote…</i></summary><blockquote>

I tried this and it seemed to fix the height of the text field, I'm not sure what other styling you were hoping to fix.
</blockquote></details>

The input field looks great! The only other thing is the font, I'm not sure why the font isn't being applied.


<!-- Sent from Reviewable.io -->

Sebastian-ubs commented:
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O23ZJ3W9uiJB1gjcfuV:b-fgguse) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

The input field looks great! The only other thing is the font, I'm not sure why the font isn't being applied.
</blockquote></details>

font inside the input field or the popover? I had added font explicitly to the dropdown menu, as the popver ends up added to the dom tree as a sibling to the rest of the app and is so not picking up the styles. maybe the same is needed here.
https://github.com/paranext/paranext-core/blob/main/lib/platform-bible-react/src/components/shadcn-ui/dropdown-menu.tsx#L119


<!-- Sent from Reviewable.io -->

dewert99 commented:
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O2B08g2CwAxCCHjh68g:by7rry1) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
<details><summary><i>Previously, Sebastian-ubs wrote…</i></summary><blockquote>

font inside the input field or the popover? I had added font explicitly to the dropdown menu, as the popver ends up added to the dom tree as a sibling to the rest of the app and is so not picking up the styles. maybe the same is needed here.
https://github.com/paranext/paranext-core/blob/main/lib/platform-bible-react/src/components/shadcn-ui/dropdown-menu.tsx#L119
</blockquote></details>

I'm not sure what to do for the font


<!-- Sent from Reviewable.io -->

jolierabideau reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: all files reviewed, 2 unresolved discussions (waiting on @dewert99 and @Sebastian-ubs)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O2Upu-0BF96pgLDwl-C:bfbwwrr) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
<details><summary><i>Previously, dewert99 (David Ewert) wrote…</i></summary><blockquote>

I'm not sure what to do for the font
</blockquote></details>

If you pull the latest main, I have a fix so that adding `pr-twp` also adds the font so you don't need to add it separately. If that doesn't work then you may need to add `pr-twp` to the `Command` component. 


<!-- Sent from Reviewable.io -->

dewert99 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970)* status: 6 of 18 files reviewed, 2 unresolved discussions (waiting on @jolierabideau)

___
*[`lib/platform-bible-react/src/components/shadcn-ui/popover.tsx` line 20 at r5](https://reviewable.io/reviews/paranext/paranext-core/970#-O1wfHPhCm1TaLIHsX74:-O2WXY6B-NbS8NdoDReS:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/1d6fb31625b371e1a83c6182ee0887e0f7870fb2/lib/platform-bible-react/src/components/shadcn-ui/popover.tsx#L20)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

If you pull the latest main, I have a fix so that adding `pr-twp` also adds the font so you don't need to add it separately. If that doesn't work then you may need to add `pr-twp` to the `Command` component.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

jolierabideau reviewed (approved):
Thanks David!!

> Reviewed 12 of 12 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/970#-:-O2WjeYBC6M0-P8O-x-x:b555jlu)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
